compositor_pipeline/src/pipeline/decoder/fdk_aac.rs

```diff
@@ -0,0 +1,202 @@
+                )
+            };
+
+            if result != fdk::AAC_DECODER_ERROR_AAC_DEC_OK {
+                return Err(AacDecoderError::FdkDecoderError(result));
+            }
+
+            let info = unsafe { *fdk::aacDecoder_GetStreamInfo(self.0) };
+
+            // The decoder should output `info.aacSamplesPerFrame` for each channel
+            let mut decoded_samples: Vec<fdk::INT_PCM> =
+                vec![0; (info.aacSamplesPerFrame * info.channelConfig) as usize];
+
+            let result = unsafe {
+                fdk::aacDecoder_DecodeFrame(
+                    self.0,
+                    decoded_samples.as_mut_ptr(),
+                    decoded_samples.len() as i32,
+                    0,
+                )
+            };
+
+            if result == fdk::AAC_DECODER_ERROR_AAC_DEC_NOT_ENOUGH_BITS {
+                // Need to put more data in
+                continue;
+            }
+
+            if result != fdk::AAC_DECODER_ERROR_AAC_DEC_OK {
+                return Err(AacDecoderError::FdkDecoderError(result));
+            }
+
+            let samples = match info.channelConfig {
+                1 => AudioSamples::Mono(decoded_samples),
+                2 => AudioSamples::Stereo(
+                    decoded_samples
+                        .chunks_exact(2)
+                        .map(|c| (c[0], c[1]))
+                        .collect(),
+                ),
+
+                _ => return Err(AacDecoderError::UnsupportedChannelConfig),
+            };
+
+            // We need the info before the decode call to get the channel info, but the sample rate
+            // can change during `aacDecoder_DecodeFrame`
+            let info = unsafe { *fdk::aacDecoder_GetStreamInfo(self.0) };
+
+            output_buffer.push(AudioSamplesBatch {
+                samples: samples.into(),
+                start_pts: chunk.pts,
+                sample_rate: info.sampleRate as u32,
+            });
+        }
+
+        Ok(output_buffer)
+    }
+}
+
+impl Drop for Decoder {
+    fn drop(&mut self) {
+        unsafe {
+            fdk::aacDecoder_Close(self.0);
+        }
+    }
+}
+
+pub struct FdkAacDecoder;
+
+impl FdkAacDecoder {
+    pub(super) fn new(
+        options: AacDecoderOptions,
+        chunks_receiver: Receiver<EncodedChunk>,
+        samples_sender: Sender<AudioSamplesBatch>,
+        input_id: InputId,
+    ) -> Result<Self, AacDecoderError> {
+        let (result_sender, result_receiver) = crossbeam_channel::bounded(1);
+
+        std::thread::Builder::new()
+            .name(format!("fdk aac decoder {}", input_id.0))
+            .spawn(move || {
+                start_decoding(
+                    options,
+                    samples_sender,
+                    chunks_receiver,
+                    input_id,
+                    result_sender,
+                )
+            })
+            .unwrap();
+
+        result_receiver.recv().unwrap()?;
+
+        Ok(Self)
+    }
+}
+
+fn start_decoding(
```

Comment:
```suggestion
fn run_decoder_thread(
```

I suggest sticking to a naming convention where
- start means that function does sth and exits quickly while sth that was started is running somewhere
- run means that function does sth in a blocking way

Reply:
Will do that in the subsequent PR